### PR TITLE
feat(ci): Add coverage and junit results for js

### PR DESCRIPTION
**Pull request by dcramer**

### CODE REVIEW

File: .babelrc.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const env = process.env.BABEL_ENV || process.env.NODE_ENV || 'development';
+
+var pluginList = [
+  env === 'development' ? ['emotion', {sourceMap: true, autoLabel: true}] : 'emotion',
+  'transform-decorators-legacy',
+  'transform-class-properties',
+  'transform-object-rest-spread',
+  'transform-runtime',
```
Comment (dcramer) [inline .babelrc.js]: without this i got weird import errors on some tests

File: .babelrc.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  'lodash',
+  'dynamic-import-node',
+  'syntax-dynamic-import',
+  'react-hot-loader/babel',
+  [
+    'babel-plugin-transform-builtin-extend',
+    {
+      globals: ['Array', 'Error'],
```
Comment (dcramer) [inline .babelrc.js]: i added Error here for no good reason other than I use it in Zeus and its nice to be able to extend it and maintain the type

File: scripts/config/env.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+'use strict';
```
Comment (dcramer) [inline scripts/config/env.js]: this file can be removed

File: scripts/test.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+process.env.TZ = 'America/New_York';
+
+// Makes the script crash on unhandled rejections instead of silently
+// ignoring them. In the future, promise rejections that are not handled will
+// terminate the Node.js process with a non-zero exit code.
+process.on('unhandledRejection', err => {
+  throw err;
+});
+
+// Ensure environment variables are read.
```
Comment (dcramer) [inline scripts/test.js]: this include can be removed

File: package.json [hunk 5]
```diff
@@ -94,18 +95,30 @@
...
     "dev-proxy": "node scripts/devproxy.js",
     "dev-server": "webpack-dev-server",
     "storybook": "start-storybook -p 9001 -c .storybook",
     "snapshot": "build-storybook && PERCY_TOKEN=$STORYBOOK_PERCY_TOKEN PERCY_PROJECT=$STORYBOOK_PERCY_PROJECT percy-storybook --widths=375,1280"
   },
   "jest": {
+    "collectCoverage": true,
+    "collectCoverageFrom": [
+      "tests/js/spec/**/*.{js,jsx}",
+      "src/sentry/static/sentry/**/*.{js,jsx}"
```
Comment (billyvg) [inline package.json]: should this be `src/sentry/static/sentry/app/**/*.{js,jsx}`?

File: package.json [hunk 6]
```diff
@@ -94,18 +95,30 @@
...
+    "jest": "node scripts/test.js --env=jsdom",
+    "jest-debug": "node --inspect-brk scripts/test.js --env=jsdom --runInBand",
     "lint": "node_modules/.bin/eslint tests/js src/sentry/static/sentry/app --ext .jsx",
     "dev-proxy": "node scripts/devproxy.js",
     "dev-server": "webpack-dev-server",
     "storybook": "start-storybook -p 9001 -c .storybook",
     "snapshot": "build-storybook && PERCY_TOKEN=$STORYBOOK_PERCY_TOKEN PERCY_PROJECT=$STORYBOOK_PERCY_PROJECT percy-storybook --widths=375,1280"
   },
   "jest": {
+    "collectCoverage": true,
```
Comment (billyvg) [inline package.json]: This should be removed, and instead we pass `--coverage` into `test-ci` tasl

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Changes to build requirements</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: package.json [hunk 7]
```diff
@@ -94,18 +95,30 @@
   "APIMethod": "stub",
   "proxyURL": "http://localhost:8000",
   "scripts": {
-    "test": "npm run jest -- tests/js/spec",
-    "test-ci": "npm run test -- --runInBand",
-    "test-watch": "npm run test -- --watch",
-    "jest": "NODE_ENV=test node_modules/.bin/jest",
-    "jest-debug": "NODE_ENV=test node --inspect-brk node_modules/.bin/jest --runInBand",
+    "test": "node scripts/test.js --env=jsdom",
+    "test-ci": "node scripts/test.js -- --runInBand --ci --testResultsProcessor=jest-junit",
```
Comment (billyvg) [inline package.json]: not that it matters much but shouldn't these all work if calling `npm run test`?

**Comment by dcramer** [comment]
dynamic-import-node being loaded outside of tests was causing weird behavior with our .less usage

**Comment by billyvg** [comment]
storybook's local presets is broken https://github.com/storybooks/storybook/issues/2635

**Review by billyvg — APPROVED** [review]

**Pull request merged**